parser/src/layout.rs

```diff
@@ -359,7 +353,10 @@ impl<'input, Tokens> Layout<'input, Tokens>
                 (&Token::With, _) => self.scan_for_next_block(Context::MatchClause)?,
 
                 (&Token::Else, _) => {
-                    let next = self.next_token();
+                    let next = match self.unprocessed_tokens.pop() {
+                        Some(token) => token,
+                        None => return Ok(None),
```

Comment:
I think this may cause the initial token (from line 148) to be forgotten? Maybe unprocessed_tokens need to be filled with enough tokens to ensure it always has tokens available (unless its at EOF).